File: src/main/java/utils/LocProperties.java
package utils;

import java.io.FileInputStream;
import java.io.IOException;
import java.util.Properties;
import java.util.logging.Level;

import utils.log.Log;

public class LocProperties {

    private static Properties properties = null;

    private LocProperties() {
        try {
            properties = new Properties();
            FileInputStream inputStream = new FileInputStream("execution.properties");
            properties.load(inputStream);
        } catch (IOException e) {
            Log.log(Level.SEVERE, "IO Exception: " + e.getMessage());
        }
    }

    public static Properties getProperties() {
        if (properties == null) {
            new LocProperties();
        }
        return properties;
    }
}

File: src/main/kotlin/utils/LocProperties.kt
package utils

import java.nio.file.Files
import java.nio.file.Paths
import java.util.Properties

class LocProperties() {

    companion object {

        val properties: Properties = Properties()

        init {
            try {
                Files.newInputStream(Paths.get("execution.properties")).use { properties.load(it) }
            } catch (e: Exception) {
                println("Fail when loading properties file: ${e.message}")
            }
        }

        fun getProperty(clave: String): String? = properties.getProperty(clave)
    }
}
